core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1594,8 +1594,29 @@ protected void updateNewComputer(Node n) {
         updateNewComputer(n, AUTOMATIC_AGENT_LAUNCH);
     }
 
+    /**
+     * Update the list of computers that are running on this Jenkins instance.
+     * Consider {@link #updateComputers(Node...)} instead if you know what nodes needs to be updated.
+     * @see #updateComputers(Node...)
+     */
     protected void updateComputerList() {
-        updateComputerList(AUTOMATIC_AGENT_LAUNCH);
+        var allNodes = new HashSet<Node>();
+        allNodes.add(this);
+        allNodes.addAll(getNodes());
+        updateComputerList(AUTOMATIC_AGENT_LAUNCH, allNodes);
+    }
+
+    /**
+     * Update the computers for the given nodes.
+     */
+    protected void updateComputers(@NonNull Node... nodes) {
+        var nodeSet = new HashSet<Node>();
+        for (var node : nodes) {
+            if (node != null) {
+                nodeSet.add(node);
+            }
```

Comment:
is it legit for someone to pass in a `null` node, seems like this would mask a bug without any warning/error?

```suggestion
            if (node != null) {
                nodeSet.add(node);
            } else {
                LOGGER.log(WARNING, "A null node was passed to updateComputers", new Throwable("null is not valid"));
            }
```

Reply:
I reworked the implementation to handle the `null` case at the caller side instead.